docs/using-docker.md

```diff
@@ -0,0 +1,38 @@
+# Using Docker for job execution
+
+## Overview 
+Turbinia can support Docker by allowing users to configure jobs to execute through a Docker container. This eliminates the need to install all the necessary dependencies and/or external programs on the host machine and instead execute the job through a container that has all the necessary dependencies installed. Please note that Turbinia does not provide the Docker images necessary for each job and they will need to either be created or pulled from a container registry. 
```

Comment:
This is relevant to your comments on the other doc that you reviewed, but in most places so far we are capitalizing `Job` and `Worker`.  Maybe we should still do that here for consistency, and separately we can review whether we should keep that practice or just lower-case everything, WDYT?